### DISCUSSION TITLE
Encoder gear teeth.

### DISCUSSION THREAD

**Discussion by kingjamez** [kachurovskiy/nanoels#4] [General]
I'm going to be building this, have started 3D printing the mounts and gears. My MX-210V lathe's headstock gear that the encoder will connect to is 56tooth. I think from a math perspective it would be best if the encoder had a 1:1 relationship with the headstock gear. Does your lathe have a 60T gear on the headstock?

**Comment by kachurovskiy** [comment]
Hi! How many lines does your encoder have? If it's 600 then it will be 56/60*600=560 value for `ENCODER_STEPS`. If it's 400 or 200 then even if you round it to the nearest number, the error on the tool is still going to be very small but probably noticeable with a micrometer.

In the code, there are a few places where there's modulo operation e.g. `% (int) ENCODER_STEPS` so I agree that having fractional encoder steps is not ideal.

It's rather easy to make a 56-tooth gear in Fusion 360, just a couple clicks in the gear generation script. What's hard is to make it fit the dimensions of your lathe without having access to it :) If you can provide a drawing with dimensions, I can help make the model for your use case. Cheers!

**Reply by kingjamez** [reply]
Hi! Thanks for the quick reply. I've got a 600ppr and a 1000ppr encoder. I've already been working on another ELS but yours is much better documented so I'm going to give it a try.

I have an MX-210 lathe, so I think everything you've done should "just work". I've already designed a 56T gear in Fusion and am printing the first version right now.
![Encoder_Gear_Lathe_56t_ELS v8](https://user-images.githubusercontent.com/11293910/106397580-39496b80-63dc-11eb-8abc-a9b4fb2efe77.png)

If it works, I can upload it if anyone wants it.

**Reply by kachurovskiy** [reply]
Sweet! Yes, definitely will be nice to have it in https://github.com/kachurovskiy/nanoels/tree/main/h1/encoder or e.g. Thingiverse where it might be easier for you upload - we can then link from the README.md to it. Thanks!

**Reply by kingjamez** [reply]
Will do. It's done printing and I'll test it on the lathe with your encoder mount tomorrow. If it works, I'll put it up on Thingiverse.

**Reply by kachurovskiy** [reply]
James, how did it go? Thank you.

**Comment by kingjamez** [comment]
It’s been more challenging than expected. With a smaller 56T gear the cone
has to be just the right size to not hit the belt drive of the spindle.
That and my 3D printer wasn’t printing well resulting in wobbly gears.

I think I have the final design done though, just waiting to print it out
and make sure it runs true at high speed on the lathe.

Otherwise, I’ve assembled almost everything else from an electronics
perspective. Still waiting on the HTD 3 belts from China to arrive, but I
should have a complete system pretty soon to begin testing.


-Jim

**Comment by kingjamez** [comment]
I've not yet run it on my lathe but am confident enough of the design to go ahead and put it up on Thingiverse. This version clears the spindle and allows the encoder to sit flush with the back of the headstock.

https://www.thingiverse.com/thing:4754021

**Reply by kachurovskiy** [reply]
Sweet! Added it to the README: https://github.com/kachurovskiy/nanoels/commit/7cee8029e4b6d33a1befe7acc17c79f86e904050
